### ISSUE TITLE
Orphaned java processes

### ISSUE THREAD

**Issue by sim642** [karoliineh/MagpieBridge-Goblint#10]
I have closed all VS Code processes and somehow the java processes from GobPie still keep running:
![image](https://user-images.githubusercontent.com/378740/138595531-91b7990a-1abe-4d4e-9f89-c7e56d2bf903.png)

I'm not sure if this is an issue in GobPie specifically (not allowing itself to be terminated for some reason) or in Magpie more generally.

**Comment by FeldrinH** [comment]
Could not reproduce on WSL by opening, running analyses and closing. Either this has been accidentally fixed or there is some particular step or environment-specific detail required to achieve this.

**Comment by sim642** [comment]
This is still happening to me on the latest version from @FeldrinH's fork at least:
![image](https://user-images.githubusercontent.com/378740/231421838-ea8b1694-ab34-41e2-a161-0c20d7297392.png)
These java processes are not children of any VS Code process, but orphaned and thus direct children of the system init process.

Somehow it's even worse now because the java processes (not their Goblint children!) are running at 100% CPU.

**Comment by sim642** [comment]
One thing that seems to cause such processes is doing "Developer: Reload Window" in VS Code. Maybe it's not handling well the closure of some LSP socket on the VS Code side.

The reason I keep having to reload window is that modifying and saving files for whatever reason stops working for me: no new analysis is ever started.

**Comment by FeldrinH** [comment]
Small update:

The 100% CPU issue *should* be fixed on https://github.com/FeldrinH/GobPie/tree/abstract-debugging and should not occur at all on GobPie versions without abstract debugging.

As for orphaned processes, I tried to reload the window and I noticed that old GobPie processes do stay open. I don't know why that is. However, when I close VSCode all GobPie processes are closed with it, so I still can't reproduce the issue itself.

Analysis-on-save breaking I cannot reproduce at all. @sim642 it would probably be a good idea to open a new issue for this, if you can reproduce it consistently on the latest version.

**Comment by FeldrinH** [comment]
Another update:

The analysis-on-save issue turned out to be that GobPie simply does not trigger an analysis when saving .h or .i files.

This should be fixed once https://github.com/goblint/GobPie/pull/58 is merged.
